### PR TITLE
Bugfix: actually use torch conv3d, additonally pass convolution_type from config

### PR DESCRIPTION
**Pull request by thomasZen**
- Fixed bug that lead to an empty core for the convolution_type == "torch" (this was introduced in https://github.com/open-retina/open-retina/pull/145)
- Unified convolution type names by using get_conv_type for GRU and conv models
- Few consistency changes
- Exposed convolution_type in the config.

### CODE REVIEW

File: openretina/models/core_readout.py [hunk 1]
```diff
@@ -144,10 +144,11 @@ def compute_readout_input_shape(
         self, core_in_shape: tuple[int, int, int, int], core: Core
     ) -> tuple[int, int, int, int]:
         # Use the same device as the core's parameters to avoid potential errors at init.
-        # device = next(core.parameters()).device
+        device = next(core.parameters()).device
```
Comment (thomasZen) [inline openretina/models/core_readout.py:147]: this was commented out in a previous commit, I think with no good reason.
But see https://github.com/open-retina/open-retina/issues/148, this runs mostly on cpu anyway as it runs before we move the weights to gpu I think.

File: openretina/modules/core/base_core.py [hunk 2]
```diff
@@ -124,48 +124,33 @@ def __init__(
...
-            elif self.convolution_type == "torch":
-                layer["conv"] = TorchSTSeparableConv3D(
-                    num_in_channels,
-                    num_out_channels,
-                    log_speed_dict={},
-                    temporal_kernel_size=temporal_kernel_sizes[layer_id],
-                    spatial_kernel_size=spatial_kernel_sizes[layer_id],
-                    bias=False,
-                    padding=padding,
-                )
```
Comment (thomasZen) [inline openretina/modules/core/base_core.py:164]: this was a bug. For the elif, self.features.add_module was never called!
This lead to a readout only model with an empty core.

File: openretina/modules/layers/convolutions.py [hunk 3]
```diff
@@ -440,3 +440,16 @@ def temporal_smoothing(sin: torch.Tensor, cos: torch.Tensor) -> torch.Tensor:
     reg = torch.sum((smoother * sin) ** 2) / F
     reg += torch.sum((smoother * cos) ** 2) / F
     return reg
+
+
+def get_conv_class(conv_type: str) -> type[nn.Module]:
```
Comment (thomasZen) [inline openretina/modules/layers/convolutions.py:453]: this function was only moved

File: openretina/modules/nonlinearities.py [hunk 4]
```diff
@@ -13,7 +13,7 @@ def forward(self, x):
         return torch.where(x > 0, self.c * x, self.a * (torch.exp(x / self.b) - 1))
 
 
-class parametrized_softplus(nn.Module):
+class ParametrizedSoftplus(nn.Module):
```
Comment (thomasZen) [inline openretina/modules/nonlinearities.py:16]: python style is to use camelCase

File: openretina/modules/core/base_core.py [hunk 5]
```diff
@@ -189,11 +174,13 @@ def temporal_laplace(self) -> torch.Tensor:
...
     def temporal_smoothness(self) -> torch.Tensor:
-        if self.convolution_type == "torch":
+        if self.convolution_type == "separable":
             return self.temporal_laplace()
-        else:
+        elif self.convolution_type == "custom_separable":
             results = [temporal_smoothing(x.conv.sin_weights, x.conv.cos_weights) for x in self.features]
             return torch.sum(torch.stack(results))
+        else:
+            raise ValueError(f"Temporal smoothness not supported for {self.convolution_type=}")
```
Comment (thomasZen) [inline openretina/modules/core/base_core.py]: I think we can implement this better in the future. Right now these losses only calculated if the loss weight > 0.0.
So it works if they the interpolation weights are set to 0.0 in the config.

We can rethink this in the future, though.

Comment (fededagos) [inline openretina/modules/core/base_core.py]: Sounds good to me. Let's maybe explicitly put this in the error then.

```suggestion
            raise ValueError(
            f"Temporal smoothness not supported for {self.convolution_type=}. "
            "Set the temporal smoothness regularisation weight to 0 to still use this conv type."
            )
```

**Review by fededagos — APPROVED** [review]
Looks good to me! 
I would maybe only make the smoothness error more explicit before merging. 
Am I correct to think that this now unifies the conv_type convention between the normal core and the gru core?

**Comment by thomasZen** [comment]
> Am I correct to think that this now unifies the conv_type convention between the normal core and the gru core?

Yes, that's true. I noticed that it was different and used that opportunity to unify it :)
